sdk/textanalytics/azure-ai-textanalytics/tests/test_analyze_healthcare.py

```diff
@@ -0,0 +1,682 @@
+
+    @GlobalTextAnalyticsAccountPreparer()
+    @TextAnalyticsClientPreparer(client_kwargs={
+        "api_version": TextAnalyticsApiVersion.V3_1_PREVIEW,
+        "text_analytics_account_key": os.environ.get('AZURE_TEXT_ANALYTICS_KEY'),
+        "text_analytics_account": "https://cognitiveusw2dev.azure-api.net"
+    })
+    def test_missing_input_records_error(self, client):
+        docs = []
+        with pytest.raises(ValueError) as excinfo:
+            client.begin_analyze_healthcare(docs)
+        assert "Input documents can not be empty or None" in str(excinfo.value)
+
+    @GlobalTextAnalyticsAccountPreparer()
+    @TextAnalyticsClientPreparer(client_kwargs={
+        "api_version": TextAnalyticsApiVersion.V3_1_PREVIEW,
+        "text_analytics_account_key": os.environ.get('AZURE_TEXT_ANALYTICS_KEY'),
+        "text_analytics_account": "https://cognitiveusw2dev.azure-api.net"
+    })
+    def test_passing_none_docs(self, client):
+        with pytest.raises(ValueError) as excinfo:
+            client.begin_analyze_healthcare(None)
+        assert "Input documents can not be empty or None" in str(excinfo.value)
+
+    @GlobalTextAnalyticsAccountPreparer()
+    @TextAnalyticsClientPreparer(client_kwargs={
+        "api_version": TextAnalyticsApiVersion.V3_1_PREVIEW,
+        "text_analytics_account_key": os.environ.get('AZURE_TEXT_ANALYTICS_KEY'),
+        "text_analytics_account": "https://cognitiveusw2dev.azure-api.net"
+    })
+    def test_duplicate_ids_error(self, client):
+        # Duplicate Ids
+        docs = [{"id": "1", "text": "hello world"},
+                {"id": "1", "text": "I did not like the hotel we stayed at."}]
+        try:
+            result = client.begin_analyze_healthcare(docs)
+        except HttpResponseError as err:
+            self.assertEqual(err.error.code, "InvalidDocument")
+            self.assertIsNotNone(err.error.message)
+
+    @GlobalTextAnalyticsAccountPreparer()
+    @TextAnalyticsClientPreparer(client_kwargs={
+        "api_version": TextAnalyticsApiVersion.V3_1_PREVIEW,
+        "text_analytics_account_key": os.environ.get('AZURE_TEXT_ANALYTICS_KEY'),
+        "text_analytics_account": "https://cognitiveusw2dev.azure-api.net"
+    })
+    def test_pass_cls(self, client):
+        def callback(pipeline_response, deserialized, _):
+            return "cls result"
+        res = client.begin_analyze_healthcare(
+            documents=["Test passing cls to endpoint"],
+            cls=callback
+        ).result()
+        assert res == "cls result"
+
+    @GlobalTextAnalyticsAccountPreparer()
+    @TextAnalyticsClientPreparer(client_kwargs={
+        "api_version": TextAnalyticsApiVersion.V3_1_PREVIEW,
+        "text_analytics_account_key": os.environ.get('AZURE_TEXT_ANALYTICS_KEY'),
+        "text_analytics_account": "https://cognitiveusw2dev.azure-api.net"
+    })
+    def test_multiple_pages_of_results_returned_successfully(self, client):
```

Comment:
OMG I LOVE YOUR TESTS!

Reply:
Thanks!  :D  I actually had to update this one for healthcare since some of the pagination code is difficult to hit due to the recent change to the service to set data limits for input, but I have similar tests for Analyze which do cover the pagination code so I think it should still be good to go.